File: backend/src/main/java/com/evalorithm/entity/QuestionStatistics.java
package com.evalorithm.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "question_statistics")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class QuestionStatistics extends BaseEntity {

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id", nullable = false)
    private Question question;

    @Column(nullable = false)
    @Builder.Default
    private Integer viewCount = 0;

    @Column(nullable = false)
    @Builder.Default
    private Integer usageCount = 0;

    @Column(nullable = false)
    @Builder.Default
    private Integer correctCount = 0;

    @Column(nullable = false)
    @Builder.Default
    private Integer wrongCount = 0;

    private LocalDateTime lastUsedAt;
}

File: android/app/src/main/java/com/evalorithm/data/model/QuestionStatistics.kt
package com.evalorithm.data.model

data class QuestionStatistics(
    val id: Long,
    val viewCount: Int,
    val usageCount: Int,
    val correctCount: Int,
    val wrongCount: Int,
    val correctPercentage: Double,
    val wrongPercentage: Double,
    val lastUsedAt: String?
)
